d3d12/.gitignore

```diff
@@ -0,0 +1,12 @@
+# Generated by Cargo
```

Comment:
Same hjere

Reply:
Removing this actually seems undesirable, since this directory is its technically own crate outside the main workspace. We happen to still ignore it because of how the `.gitignore` matching rules work (i.e., `target/` in our top-level `.gitignore` matches a `target` folder in _all_ subdirectories 😶), but this `.gitignore` still makes sense, IMO.

I'll still delete it if you want, but I don't think this is the same class of redundancy as the `rustfmt.toml` or GH actions configuration.